### ISSUE TITLE
Get Key failed: password has been cleared (PKCS12, SunJSSE compat) yet again

### ISSUE THREAD

**Issue by dslaveykov** [bcgit/bc-java#805]
Hello, 
using bcprov-jkd15on, version 1.65 or later
and java 11.0.6 or later

I still have exactly the same issue as this one ->  #586 

A simple test to reproduce:

```
public static void main(String[] args) throws Exception {

        String keyStoreFile = "keystoreFile";
        String keystorePassword = "keystorePassword";
        String keyAlias = "keyAlias";
        KeyStore keystore = KeyStore.getInstance("JKS");
        keystore.load(new FileInputStream(keyStoreFile), keystorePassword.toCharArray());
        PrivateKey privateKey = (PrivateKey) keystore.getKey(keyAlias, null);

}
```

Is this actually fixed?
Currently I can not use other version of bc-prov or java, so I'm stuck this problem.

Regards.

**Comment by ddieterly** [comment]
Also having this same issue. 

|    |    \--- org.bouncycastle:bcpg-jdk15on:1.61
|    |         \--- org.bouncycastle:bcprov-jdk15on:1.61

```Caused by: java.security.UnrecoverableKeyException: failed to decrypt safe contents entry: java.lang.IllegalStateException: password has been cleared
```

**Comment by dghgit** [comment]
I guess there's two things.

The initial issue isn't actually using BC, although I should also point out that last I looked passing null to a JKS keystore as a password was not allowed and caused an exception.

The second question appears to relate to 1.61 and #586 as the issue states, the problem was fixed in 1.65.
